tests/fuzz/s2n_client_hello_recv_fuzz_test.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_config.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_crypto.h"
+#include "tls/s2n_tls.h"
+#include "tls/s2n_tls_parameters.h"
+#include "utils/s2n_random.h"
+#include "utils/s2n_safety.h"
+#include "s2n_test.h"
+
+static const uint8_t TLS_VERSIONS[] = {S2N_TLS10, S2N_TLS11, S2N_TLS12};
+
+int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    for(int version = 0; version < sizeof(TLS_VERSIONS); version++){
```

Comment:
nit: The conditional `version < sizeof(TLS_VERSIONS)` works, but only because we we're using a uint8_t array. `version < (sizeof(TLS_VERSION) / TLS_VERSIONS[0])` might make what we're doing here more clear(get the number of elements in the array). It's also defends against a future developer blindly copying this loop and assuming it will work for any array. What do you think?